_posts/AzerothCoreHostedTutorial.md

```diff
@@ -0,0 +1,144 @@
+- Let's refresh the permissions for MariaDB `Flush Privileges;`
+- Return to the main Debian console `exit`
+
+#### Configure Remote Connections to MariaDB
+- Navigate to the folder holding the file we need `cd /etc/mysql/mariadb.conf.d`
+- Launch the nano file editor for the configuration file `nano 50-server.cnf`
+- Navigate to the Bind Address, and update it to: `0.0.0.0`
+- To save your changes, press *Ctrl + X*, then *Y*, followed by *Ctrl + T*. Then use your arrow keys to select `50-server.cnf`
+- Now that we've updated a key MariaDB Configuration file, we need to restart the MariaDB process. Enter `sudo /etc/init.d/mysql restart`
+
+#### Configure the Uncomplicated Firewall (UFW)
+- Let's refresh our catalog of apps `sudo apt update`
+- Let's install UFW `sudo apt install ufw`
+- Next, we're going to blanket-block all Incoming connections. This is an important step to protecting your server. We'll open up ports in the following steps `sudo ufw default deny incoming`
+- Next, let's blanket-allow all Outgoing connections. This should allow your server to make callouts without issue. `sudo ufw default allow outgoing`
+- We need to make sure we don't get locked out of the Console, so we need to unblock port 22, which is the port SSH uses. To do this, enter `sudo ufw allow ssh`
+- Let's allow the MariaDB Port to accept incoming connections `sudo ufw allow 3306`
+- Let's enable the Authentication Server to accept incoming connections `sudo ufw allow 3724`
+- And now the Worldserver port `sudo ufw allow 8085`
+- Verify that you've enabled all the above ports, and then turn on UFW with this: `sudo ufw enable`
+
+## AzerothCore Installation
+#### Let's clone the AzerothCore Git
+- Log in as our sudo user `sudo su azcore`
+- Navigate to our home directory `cd` 
+- Let's download and clone the latest version of AzerothCore `git clone https://github.com/azerothcore/azerothcore-wotlk.git --branch master --single-branch azerothcore`
+- Once the clone is complete, let's navigate to the top level folder of the git directory `cd azerothcore`
+- We need to make a folder called *build* `mkdir build`
+- Navigate to the new build folder `cd build`
+- Now we run the cmake command, this is the pre-compile step to ensure all cpp files are accounted for before we compile, and it tells the compile what to compile. `cmake ../ -DCMAKE_INSTALL_PREFIX=$HOME/azeroth-server/ -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DTOOLS=0 -DSCRIPTS=1`
+- Now we compile AzerothCore - this can take some time, depending on the number of cpu cores your Droplet has. This tutorial is based on the 4 cpu core Droplet, so we'll use `make -j 4`. If you've got a different number of cores, replace the 4 with the number of cpu cores you have.
```

Comment:
we have a generic formula that is numcores + 2, it can be generalized using something like:

```
MTHREADS=`grep -c ^processor /proc/cpuinfo`; MTHREADS=$(($MTHREADS + 2));
make -j $MTHREADS;
make install -j $MTHREADS;
```